### PR TITLE
Sort querystring prior to signing

### PR DESCRIPTION
**Pull request by onno-vos-dev**
Without this fix, multipart uploading is still broken. This was not caught in previous tests as s3mock from Adobe apparently allows our request to go through without sorted querystring upon signing. In our staging and production system we only tested the `create_multipart` part of the multipart uploading flow without going all the way to the end.

Added a test that verifies that querystring must always be sorted.

### CODE REVIEW

**Comment by onno-vos-dev** [comment]
Add a test

**Review by anha0825 — APPROVED** [review]

File: src/aws_request.erl [hunk 1]
```diff
@@ -197,17 +197,17 @@ canonical_request(Method, URL, Headers, Body) ->
 %% URL and query string as separate values.
 split_url(URL) ->
     URI = hackney_url:parse_url(URL),
-    %% FIXME(jkakar) Query string name/value pairs should be URL encoded
```
Comment (onno-vos-dev) [inline src/aws_request.erl:200]: This FIXME is what bit us in the end.

**Review by enidgjoleka — APPROVED** [review]

File: src/aws_request.erl (hunk 1 above)
Comment (jkakar) [reply src/aws_request.erl:200]: Oops, sorry about that! ❤️

**Review by jkakar — APPROVED** [review]
Thanks for this!

**Comment by onno-vos-dev** [comment]
@jkakar I lack merge rights, any chance you can merge it for me? 😊

**Comment by onno-vos-dev** [comment]
Thank you @jkakar <3

**Pull request merged**